src/Seq.Extensions.Logging/Serilog/Extensions/Logging/SerilogLoggerProvider.cs

```diff
@@ -44,44 +45,42 @@ public FrameworkLogger CreateLogger(string name)
 
         public IDisposable BeginScope<T>(T state)
         {
-            return new SerilogLoggerScope(this, state);
+            return _scopeProvider.Push(state);
         }
 
         /// <inheritdoc />
         public void Enrich(LogEvent logEvent, ILogEventPropertyFactory propertyFactory)
         {
-            List<LogEventPropertyValue> scopeItems = null;
-            for (var scope = CurrentScope; scope != null; scope = scope.Parent)
+            List<LogEventPropertyValue> scopeItems = new List<LogEventPropertyValue>();
+
```

Comment:
Unfortunate allocation, but found no way to pass this by reference to the scope enumerating delegate.